File: src/history.cpp
#include "history.h"

History::History() {
}

void History::addToHistory(Field next){
	hist.append(next);
}

Field History::takeFromHistory(){
    if(notEmpty()==FALSE){
        throw new EmptyHistory();
    }
	return hist.takeLast();

}

void History::cleanHistory(){
	hist.clear();
}

bool History::notEmpty(){
	if(hist.isEmpty()){
		return FALSE;
	}
	return TRUE;
}

int History::countHistory() {
	return hist.count();
}

File: java/history/History.java
package history;

import java.util.ArrayList;
import java.util.List;

import field.Field;

public class History {
	private List<Field> hist;

	public History() { hist = new ArrayList<Field>(); }
	
	public void addToHistory(Field next) {
		hist.add(next);
	}
	public Field takeFromHistory() {
		return hist.remove(hist.size()-1);
	}
	public void cleanHistory() {
		hist.clear();
	}
	public boolean notEmpty() {
		if (hist.isEmpty())
			return false;
		return true;
	}
}
